Description: make --lock expect a reply so the screen gets properly locked
 when gnome-screensaver needs to be respawned by dbus. This was preventing
 ctrl-alt-l from locking the screen on the first attempt when the
 gnome-screensaver process was dead.
Author: Marc Deslauriers <marc.deslauriers@canonical.com>
Forwarded: no, upstream doesn't configure gnome-screensaver as a dbus service

Index: gnome-screensaver-3.2.0/src/gnome-screensaver-command.c
===================================================================
--- gnome-screensaver-3.2.0.orig/src/gnome-screensaver-command.c	2012-02-09 21:27:09.936449525 -0500
+++ gnome-screensaver-3.2.0/src/gnome-screensaver-command.c	2012-02-09 21:32:41.140458005 -0500
@@ -260,8 +260,12 @@
         }
 
         if (do_lock) {
-                reply = screensaver_send_message_void (connection, "Lock", FALSE);
-                g_assert (reply == NULL);
+                reply = screensaver_send_message_void (connection, "Lock", TRUE);
+                if (reply == NULL) {
+                        g_message ("Did not receive a reply from the screensaver.");
+                        goto done;
+                }
+                g_object_unref (reply);
         }
 
         if (do_activate) {
